packages/polymorphic/src/InferredPolymorphic.tsx

```diff
@@ -0,0 +1,37 @@
+import { usePolymorphic } from './Polymorphic.hooks';
+import { AnchorProps, PolymorphicAs } from './Polymorphic.types';
+
+/**
+ * Wrapping props in this type ensures that if `href` is defined,
+ * the `as` type is set to `'a'`, and all anchor props are accepted
+ */
+export type InferredPolymorphicProps<P = {}> =
```

Comment:
this should prob live in the types file

Reply:
I kept this one here to keep all the "inferred" utilities together

Reply:
cc @spark33

Reply:
I think in this case, we should have a `InferredPolymorphic` subdir with a types.ts file inside. I could also see each of the other functions being their own files.